gnomad/utils/constraint.py

```diff
@@ -519,17 +519,22 @@ def build_models(
         mu_snp_expr=high_cov_group_ht.mu_snp,
         observed_variants_expr=high_cov_group_ht.observed_variants,
         possible_variants_expr=high_cov_group_ht.possible_variants,
-        pop_observed_variants_array_expr=[
+        pops_observed_variants_array_expr=[
             high_cov_group_ht[f"observed_{pop}"] for pop in pops
         ],
         weighted=weighted,
     )
     _plateau_models = dict(
         high_cov_group_ht.aggregate(hl.struct(**plateau_models_agg_expr))
     )
-    # Map the models to their corresponding populations if pops is specified.
-    pop_models = _plateau_models["pop"]
-    plateau_models = {pop: hl.literal(pop_models[idx]) for idx, pop in enumerate(pops)}
+
+    pop_models = {}
+    if pops:
+        # Map the models to their corresponding populations if pops is specified.
+        pop_models = _plateau_models["pop"]
+        plateau_models = {
+            pop: hl.literal(pop_models[idx]) for idx, pop in enumerate(pops)
+        }
     plateau_models["total"] = _plateau_models["total"]
     plateau_models = hl.struct(**plateau_models)
```

Comment:
these two lines (with 527-529 above) aggregate the plateau models into a struct, convert them into a dictionary, and then reformats them again into a struct, which is a bit redundant. using an if/else statement here (while not as clean) would avoid these unnecessary reformats